test/integration/smoke/test_activate_template.py

```diff
@@ -0,0 +1,231 @@
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    def setUp(self):
+        # Default URL
+        self.getDefaultURLCmd = getSystemVMTemplateDefaultUrl.getSystemVMTemplateDefaultUrlCmd()
+        self.getDefaultURLCmd.hypervisor = self.hypervisor
+        self.urlResponse = self.apiclient.getSystemVMTemplateDefaultUrl(self.getDefaultURLCmd)
+        
+        # Official cloudstack system vm template
+        self.test_template = registerTemplate.registerTemplateCmd()
+        self.test_template.hypervisor = self.hypervisor
+        self.test_template.zoneid = self.zone.id
+        self.test_template.name = 'test-system-' + self.hypervisor + '-4.11.3'
+        self.test_template.displaytext = 'test-system-' + self.hypervisor + '-4.11.3'
+        self.test_template.url = self.urlResponse.url.url
+        self.test_template.format = self.urlResponse.url.type
+        self.test_template.system = True
+        self.test_template.ostypeid = self.getOsType("Debian GNU/Linux 9 (64-bit)")
+        self.md5 = "d40bce40b2d5bb4ba73e56d1e95aeae5"
+
+        self.activateTemplateCmd = activateSystemVMTemplate.activateSystemVMTemplateCmd()
+        self.listConfigurationsCmd = listConfigurations.listConfigurationsCmd()
+        self.listConfigurationsCmd.name = 'router.template.' + self.hypervisor
+        return
+
+    @attr(tags=["advanced", "smoke"], required_hardware="true")
+    def test_01_activate_sytem_vm_template(self):
+        """
+        Test activating registered template
+        """
+        template = self.registerTemplate(self.test_template)
+        # Registering and Downloading System VM Template
+        self.download(self.apiclient, template.id)
+        self.activateTemplateCmd.id = template.id
+        # Activating downloaded template
+        self.activateTemplate(self.activateTemplateCmd)
+        response = self.checkConfiguration(self.listConfigurationsCmd)
+        # Checking template activation
+        self.assertEqual(response[0].value, "test-system-"+ self.hypervisor + "-4.11.3", "Expected template name to be test-system-"+ self.hypervisor +"-4.11.3")
+        return
+
+    @attr(tags=["advanced", "smoke"], required_hardware="true")
+    def test_02_get_default_url(self):
+        response = self.apiclient.getSystemVMTemplateDefaultUrl(self.getDefaultURLCmd)
+        self.assertNotEqual(response.url.url, "", "Expected a value")
+        # This test will have to be updated for every official template version
+        if self.hypervisor == "kvm":
+            self.assertEqual(response.url.url, "https://download.cloudstack.org/systemvm/4.11/systemvmtemplate-4.11.3-kvm.qcow2.bz2", "Default url mismatch")
+
+    @attr(tags=["advanced", "smoke"], required_hardware="true")
+    def test_03_seedSystemVMTemplate(self):
+        activeTemplate = self.apiclient.listTemplates(self.listTemplatesCmd)[0]
```

Comment:
no need of this variable, cmd.templateid later can call directly the listTemplates api, and what if we have multiple active templates, how do we know which one we're dealing with? Probably include the given hypervisor.